pkgs/development/python-modules/lintrunner/default.nix

```
@@ -0,0 +1,55 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  maturin,
+  rustPlatform,
+  cargo,
+  rustc,
+  python,
+}:
+
+buildPythonPackage rec {
+  pname = "lintrunner";
+  version = "0.12.7";
+  format = "pyproject";
+
+  # NOTE: there is a PR open on the upstream repository for the Cargo.lock file.
+  # After that is merged we can change the owner and rev to the upstream
+  # repository instead of a fork.
+  src = fetchFromGitHub {
+    owner = "PrestonHager";
+    repo = pname;
+    rev = "main";
+    #rev = "v${version}";
+    hash = "sha256-CYSNEDT9RO7bnVW8zcDDJOva4GOKGMNs1CcU//IbBrc=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    hash = "sha256-dv/CznOB/VFx1PM+cYmdHuiK7d8z/HUQbDt5iy5LUTI=";
+  };
+
+  # Adding maturninBuildHook will automatically build and install the executable
+  nativeBuildInputs = [
+    maturin
+    rustPlatform.cargoSetupHook
+    rustPlatform.maturinBuildHook
+    cargo
+    rustc
+  ];
+
+  meta = with lib; {
+    description = "A tool that runs linters.";
```

```suggestion
    description = "Tool that runs linters";
```
https://nixos.org/manual/nixpkgs/unstable/#var-meta-description